packages/babel-parser/src/plugins/placeholders.js

```diff
@@ -229,8 +229,10 @@ export default (superClass: Class<Parser>): Class<Parser> =>
     ): T {
       const type = isStatement ? "ClassDeclaration" : "ClassExpression";
 
-      this.next();
-      this.takeDecorators(node);
+      if (this.eat(tt._class)) {
```

Comment:
With this change we will incorrectly allow
```js
class class A {}
```
because `class` is eaten twice.

I suggest we remove `this.takeDecorators` in `parseClassOrClassHasInstanceExpression` and remove `this.next()` above.